CentOS 7 build failure: "seccomp.go:161:4: error: 'EM_ARM' undeclared here (not in a function)"

# Required information

* Distribution: CentOS
* Distribution version: 7
* Architecture: x86_64

Build environment:
 * golang: 1.11.5-1.el7
 * glibc-devel: 2.17-260.el7_6.6.x86_64
 * kernel-headers: 3.10.0-957.21.3.el7.x86_64
 * libacl-devel: 2.2.51-14.el7
 * libcap-devel: 2.22-9.el7
 * libseccomp-devel: 2.3.1-3.el7
 * libuv-devel: 1:1.30.1-1.el7
 * lxc-devel: 3.1.0-0.3.el7

# Issue description

Building lxd-3.15 fails with the following error:
```
cd /builddir/build/BUILD/lxd-3.15/_output/src/github.com/lxc/lxd/lxd                                                                                                                                                                          
pkg-config --cflags -- lxc                                                                                             
pkg-config --libs -- lxc                                                                                               
CGO_LDFLAGS='"-L/builddir/build/BUILD/lxd-3.15/dist/sqlite/.libs/" "-L/builddir/build/BUILD/lxd-3.15/dist/libco/" "-L/builddir/build/BUILD/lxd-3.15/dist/raft/.libs/" "-L/builddir/build/BUILD/lxd-3.15/dist/dqlite/.libs/" "-Wl,-rpath,/usr/l
ib64/lxd" "-lcap" "-llxc" "-llxc" "-lutil"' /usr/lib/golang/pkg/tool/linux_amd64/cgo -objdir $WORK/b001/ -importpath github.com/lxc/lxd/lxd -- -I $WORK/b001/ -I/builddir/build/BUILD/lxd-3.15/dist/sqlite/ -I/builddir/build/BUILD/lxd-3.15/d
ist/libco/ -I/builddir/build/BUILD/lxd-3.15/dist/raft/include/ -I/builddir/build/BUILD/lxd-3.15/dist/dqlite/include/ -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -s
td=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla -std=gnu11 -Wvla ./main_checkfeature.go ./main_forkdns.go ./main_forkfile.go ./main_forkmknod.go ./main_forkmount.go ./main_forknet.go ./main_forkproxy.go ./main_forkuevent.go ./main_nsexec
.go ./seccomp.go ./storage_cgo.go            
# github.com/lxc/lxd/lxd                     
In file included from _output/src/github.com/lxc/lxd/lxd/seccomp.go:141:0:                                             
_output/src/github.com/lxc/lxd/lxd/seccomp.go:161:4: error: 'EM_ARM' undeclared here (not in a function)               
  { AUDIT_ARCH_ARM,          14, 324 },                   
    ^                                                                                                                  
_output/src/github.com/lxc/lxd/lxd/seccomp.go:200:4: error: 'EM_OPENRISC' undeclared here (not in a function)          
  { AUDIT_ARCH_OPENRISC,     -1,  33 },
    ^                                                                                                                  
_output/src/github.com/lxc/lxd/lxd/seccomp.go:224:4: error: 'EM_MICROBLAZE' undeclared here (not in a function)                                                                                                                               
  { AUDIT_ARCH_MICROBLAZE,   14, 297 },                                                                                                                                                                                                       
    ^
```

--- a/lxd/seccomp.go
+++ b/lxd/seccomp.go
@@ -31,6 +31,7 @@ import (
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE 1
 #endif
+#include <elf.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <linux/seccomp.h>
@@ -176,12 +177,6 @@ static const struct lxd_seccomp_data_arch seccomp_notify_syscall_table[] = {
 #ifdef AUDIT_ARCH_S390X
 	{ AUDIT_ARCH_S390X,        14, 290, 224 },
 #endif
-#ifdef AUDIT_ARCH_RISCV32
-	{ AUDIT_ARCH_RISCV32,      -1,  33,   5 },
-#endif
-#ifdef AUDIT_ARCH_RISCV64
-	{ AUDIT_ARCH_RISCV64,      -1,  33,   5 },
-#endif
 #ifdef AUDIT_ARCH_PPC
 	{ AUDIT_ARCH_PPC,          14, 288, 209 },
 #endif
@@ -191,44 +186,11 @@ static const struct lxd_seccomp_data_arch seccomp_notify_syscall_table[] = {
 #ifdef AUDIT_ARCH_PPC64LE
 	{ AUDIT_ARCH_PPC64LE,      14, 288, 209 },
 #endif
-#ifdef AUDIT_ARCH_IA64
-	{ AUDIT_ARCH_IA64,         13, 259, 193 },
-#endif
 #ifdef AUDIT_ARCH_SPARC
 	{ AUDIT_ARCH_SPARC,        14, 286, 169 },
 #endif
 #ifdef AUDIT_ARCH_SPARC64
 	{ AUDIT_ARCH_SPARC64,      14, 286, 169 },
-#endif
-#ifdef AUDIT_ARCH_ALPHA
-	{ AUDIT_ARCH_ALPHA,        14, 452, 382 },
-#endif
-#ifdef AUDIT_ARCH_OPENRISC
-	{ AUDIT_ARCH_OPENRISC,     -1,  33,   5 },
-#endif
-#ifdef AUDIT_ARCH_PARISC
-	{ AUDIT_ARCH_PARISC,       14, 277, 238 },
-#endif
-#ifdef AUDIT_ARCH_PARISC64
-	{ AUDIT_ARCH_PARISC64,     14, 277, 238 },
-#endif
-#ifdef AUDIT_ARCH_CRIS
-	{ AUDIT_ARCH_CRIS,         -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_CSKY
-	{ AUDIT_ARCH_CSKY,         -1,  33,   5 },
-#endif
-#ifdef AUDIT_ARCH_FRV
-	{ AUDIT_ARCH_FRV,          -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_M32R
-	{ AUDIT_ARCH_M32R,         -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_M68K
-	{ AUDIT_ARCH_M68K,         14, 290, 223 },
-#endif
-#ifdef AUDIT_ARCH_MICROBLAZE
-	{ AUDIT_ARCH_MICROBLAZE,   14, 297, 226 },
 #endif
 #ifdef AUDIT_ARCH_MIPS
 	{ AUDIT_ARCH_MIPS,         14, 290, 224 },
@@ -247,30 +209,6 @@ static const struct lxd_seccomp_data_arch seccomp_notify_syscall_table[] = {
 #endif
 #ifdef AUDIT_ARCH_MIPSEL64N32
 	{ AUDIT_ARCH_MIPSEL64N32, 131, 253, 180 },
-#endif
-#ifdef AUDIT_ARCH_SH
-	{ AUDIT_ARCH_SH,           14, 297, 226 },
-#endif
-#ifdef AUDIT_ARCH_SHEL
-	{ AUDIT_ARCH_SHEL,         14, 297, 226 },
-#endif
-#ifdef AUDIT_ARCH_SH64
-	{ AUDIT_ARCH_SH64,         14, 325, 254 },
-#endif
-#ifdef AUDIT_ARCH_SHEL64
-	{ AUDIT_ARCH_SHEL64,       14, 325, 254 },
-#endif
-#ifdef AUDIT_ARCH_TILEGX
-	{ AUDIT_ARCH_TILEGX,       -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_TILEGX32
-	{ AUDIT_ARCH_TILEGX32,     -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_TILEPRO
-	{ AUDIT_ARCH_TILEPRO,      -1,  -1,  -1 },
-#endif
-#ifdef AUDIT_ARCH_XTENSA
-	{ AUDIT_ARCH_XTENSA,       36, 290,  68 },
 #endif
 };
 

